File: goland/estatisticas_numeros.go
package main

import (
    "fmt"
)

func main() {
    var num, soma, qtd, pares, somaPares, impares int
    maior := -99999
    menor := 99999
    var media, mediaPares, porcentagemImpares float64

    fmt.Println("Digite números inteiros (100000 encerra):")

    for {
        fmt.Print("Número: ")
        fmt.Scan(&num)

        if num == 100000 {
            break
        }

        soma += num
        qtd++

        if num > maior {
            maior = num
        }
        if num < menor {
            menor = num
        }

        if num%2 == 0 {
            pares++
            somaPares += num
        } else {
            impares++
        }
    }

    if qtd > 0 {
        media = float64(soma) / float64(qtd)
        fmt.Printf("Soma: %d\n", soma)
        fmt.Printf("Quantidade: %d\n", qtd)
        fmt.Printf("Média: %.2f\n", media)
        fmt.Printf("Maior número: %d\n", maior)
        fmt.Printf("Menor número: %d\n", menor)

        if pares > 0 {
            mediaPares = float64(somaPares) / float64(pares)
            fmt.Printf("Média dos pares: %.2f\n", mediaPares)
        } else {
            fmt.Println("Nenhum número par digitado.")
        }

        porcentagemImpares = (float64(impares) * 100) / float64(qtd)
        fmt.Printf("Porcentagem de ímpares: %.2f%%\n", porcentagemImpares)
    } else {
        fmt.Println("Nenhum número válido digitado.")
    }
}

File: dart/estatisticas_numeros.dart
import 'dart:io';

void main() {
  int soma = 0, qtd = 0, pares = 0, somaPares = 0, impares = 0;
  int maior = -99999, menor = 99999;

  print('Digite números inteiros (100000 encerra):');

  while (true) {
    stdout.write('Número: ');
    String? input = stdin.readLineSync();
    if (input == null) continue;

    int num = int.tryParse(input) ?? 0;

    if (num == 100000) break;

    soma += num;
    qtd++;

    if (num > maior) maior = num;
    if (num < menor) menor = num;

    if (num % 2 == 0) {
      pares++;
      somaPares += num;
    } else {
      impares++;
    }
  }

  if (qtd > 0) {
    double media = soma / qtd;
    print('Soma: $soma');
    print('Quantidade: $qtd');
    print('Média: ${media.toStringAsFixed(2)}');
    print('Maior número: $maior');
    print('Menor número: $menor');

    if (pares > 0) {
      double mediaPares = somaPares / pares;
      print('Média dos pares: ${mediaPares.toStringAsFixed(2)}');
    } else {
      print('Nenhum número par digitado.');
    }

    double porcentagemImpares = (impares * 100) / qtd;
    print('Porcentagem de ímpares: ${porcentagemImpares.toStringAsFixed(2)}%');
  } else {
    print('Nenhum número válido digitado.');
  }
}
